src/mysql.erl

```diff
@@ -731,3 +743,39 @@ query_call(Conn, CallReq) ->
     end.
 
 
+-spec valid_params([term()]) -> boolean().
+valid_params(Values) when is_list(Values) ->
+    lists:all(fun is_valid_param/1, Values).
+
+-spec is_valid_param(term()) -> boolean().
+is_valid_param(null) ->
+    true;
+is_valid_param(Value) when is_binary(Value) ->
+    true;
+is_valid_param(Value) when is_list(Value) ->
+    try
+        unicode:characters_to_binary(Value)
+    of
+        Value1 when is_binary(Value1) ->
+            true;
+        _ ->
```

Comment:
I think this is the only way and thus the right way to do unicode validation.

The catch-all variable here could called `_ErrorOrIncomplete` instead of just `_`, for clarity.

Reply:
Ah, good idea :) I moved it `valid_params` into `mysql_protocol` and set up a test for it there. This way, it is closer to the related `encode_param` function, too.

I also renamed the catch-all variable as you suggested.